ESP32-C3 Support

```
ESP32-C3
RISC-V processor (RV32IMC)
Arduino 1.8.13
IRremoteESP8266 library 2.8.0
```
Is it possible to support ESP32-C3?
Example `IRrecvDemo.ino` does not compile

```
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp: In member function 'uint32_t IRrecv::ticksLow(uint32_t, uint8_t, uint16_t)':
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1082:8: error: no matching function for call to 'max(int32_t, int)'
       0));
        ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:61,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:219:5: note: candidate: 'template<class _Tp> const _Tp& std::max(const _Tp&, const _Tp&)'
     max(const _Tp& __a, const _Tp& __b)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:219:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1082:8: note:   deduced conflicting types for parameter 'const _Tp' ('long int' and 'int')
       0));
        ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:61,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:265:5: note: candidate: 'template<class _Tp, class _Compare> const _Tp& std::max(const _Tp&, const _Tp&, _Compare)'
     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:265:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1082:8: note:   deduced conflicting types for parameter 'const _Tp' ('long int' and 'int')
       0));
        ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:62,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3462:5: note: candidate: 'template<class _Tp> _Tp std::max(std::initializer_list<_Tp>)'
     max(initializer_list<_Tp> __l)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3462:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1082:8: note:   mismatched types 'std::initializer_list<_Tp>' and 'long int'
       0));
        ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:62,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3468:5: note: candidate: 'template<class _Tp, class _Compare> _Tp std::max(std::initializer_list<_Tp>, _Compare)'
     max(initializer_list<_Tp> __l, _Compare __comp)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3468:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1082:8: note:   mismatched types 'std::initializer_list<_Tp>' and 'long int'
       0));
        ^
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp: In member function 'bool IRrecv::matchAtLeast(uint32_t, uint32_t, uint8_t, uint16_t)':
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1165:75: error: no matching function for call to 'min(uint32_t&, unsigned int)'
   return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
                                                                           ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:61,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:195:5: note: candidate: 'template<class _Tp> const _Tp& std::min(const _Tp&, const _Tp&)'
     min(const _Tp& __a, const _Tp& __b)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1165:75: note:   deduced conflicting types for parameter 'const _Tp' ('long unsigned int' and 'unsigned int')
   return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
                                                                           ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:61,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:243:5: note: candidate: 'template<class _Tp, class _Compare> const _Tp& std::min(const _Tp&, const _Tp&, _Compare)'
     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algobase.h:243:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1165:75: note:   deduced conflicting types for parameter 'const _Tp' ('long unsigned int' and 'unsigned int')
   return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
                                                                           ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:62,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3450:5: note: candidate: 'template<class _Tp> _Tp std::min(std::initializer_list<_Tp>)'
     min(initializer_list<_Tp> __l)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3450:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1165:75: note:   mismatched types 'std::initializer_list<_Tp>' and 'long unsigned int'
   return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
                                                                           ^
In file included from c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\algorithm:62,
                 from C:\Users\Alex1\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.1\cores\esp32/Arduino.h:155,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.h:10,
                 from C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:6:
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3456:5: note: candidate: 'template<class _Tp, class _Compare> _Tp std::min(std::initializer_list<_Tp>, _Compare)'
     min(initializer_list<_Tp> __l, _Compare __comp)
     ^~~
c:\users\alex1\appdata\local\arduino15\packages\esp32\tools\riscv32-esp-elf-gcc\gcc8_4_0-esp-2021r2\riscv32-esp-elf\include\c++\8.4.0\bits\stl_algo.h:3456:5: note:   template argument deduction/substitution failed:
C:\Users\Alex1\Documents\Arduino\libraries\IRremoteESP8266\src\IRrecv.cpp:1165:75: note:   mismatched types 'std::initializer_list<_Tp>' and 'long unsigned int'
   return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
                                                                           ^
exit status 1
Error compiling for board ESP32C3 Dev Module.
```

--- a/src/IRrecv.cpp
+++ b/src/IRrecv.cpp
@@ -1079,7 +1079,7 @@ uint32_t IRrecv::ticksLow(const uint32_t usecs, const uint8_t tolerance,
   // max() used to ensure the result can't drop below 0 before the cast.
   return ((uint32_t)std::max(
       (int32_t)(usecs * (1.0 - _validTolerance(tolerance) / 100.0) - delta),
-      0));
+      (int32_t)0));
 }
 
 /// Calculate the upper bound of the nr. of ticks.
@@ -1162,7 +1162,8 @@ bool IRrecv::matchAtLeast(uint32_t measured, uint32_t desired,
   // We really should never get a value of 0, except as the last value
   // in the buffer. If that is the case, then assume infinity and return true.
   if (measured == 0) return true;
-  return measured >= ticksLow(std::min(desired, MS_TO_USEC(params.timeout)),
+  return measured >= ticksLow(std::min(desired,
+                                       (uint32_t)MS_TO_USEC(params.timeout)),
                               tolerance, delta);
 }
 

